sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -720,25 +720,14 @@ static void _receive(gnrc_pktsnip_t *pkt)
         iface = ((gnrc_netif_hdr_t *)netif->data)->if_pid;
     }
 
-    if ((pkt->next != NULL) && (pkt->next->type == GNRC_NETTYPE_IPV6) &&
-        (pkt->next->size == sizeof(ipv6_hdr_t))) {
-        /* IP header was already marked. Take it. */
-        ipv6 = pkt->next;
-
-        if (!ipv6_hdr_is(ipv6->data)) {
-            DEBUG("ipv6: Received packet was not IPv6, dropping packet\n");
-            gnrc_pktbuf_release(pkt);
-            return;
-        }
-#ifdef MODULE_GNRC_IPV6_WHITELIST
-        if (!gnrc_ipv6_whitelisted(&((ipv6_hdr_t *)(ipv6->data))->src)) {
-            DEBUG("ipv6: Source address not whitelisted, dropping packet\n");
-            gnrc_pktbuf_release(pkt);
-            return;
+    for (ipv6 = pkt; ipv6 != NULL; ipv6 = ipv6->next) { /* find IPv6 header if already marked */
```

Comment:
>No, since I search for three conditions (one not directly related to a value in the node struct).

couldn't we just reduce the conditions to `ipv6->type == GNRC_NETTYPE_IPV6` and transform the other ones into asserts?
The ipv6 packet size and the header type should be fine if the functions are used that `gnrc` provides to build ipv6 packets.

Besides of that, we also used `LL_SEARCH_SCALAR` for the netif header above. This would also required the ckecks for the packet size, but we omitted them..